src/containers/MemoryGameContainer/MemoryGameContainer.jsx

```diff
@@ -0,0 +1,45 @@
+import React, { useCallback } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { BrowserRouter, Route } from 'react-router-dom';
+
+import MainContainer from '../MainContainer';
+import WelcomePage from '../../components/WelcomPage';
+import { setPlayer } from './playFormActions';
+import PlayerForm from '../../components/PlayerForm';
+
+export default function MemoryGameContainer() {
+  const player = useSelector((state) => state.playFormReducer.player);
+  const dispatch = useDispatch();
+
+  const savePlayer = useCallback((e) => {
+    e.preventDefault();
+    const inputs = [...e.currentTarget.elements].slice(0, 3);
```

Comment:
slice creates copy by default, so you don't need copy elements before slice them